enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,320 @@
+customer technician.
+
+Eventually, the cluster will need to be upgraded, and then the technician will need a supported and
+documented procedure to do it.
+
+### User stories
+
+#### Prepare an upgrade package
+
+As an engineer working in a partner's factory, I want to be able to assemble the upgrade package
+that will be delivered to the technicians that will perform the upgrade on-site.
+
+#### Upgrade a single-node cluster
+
+As a technician, I want to be able to upgrade a single node cluster using the upgrade package and
+without requiring/deploying an image registry server neither in the cluster itself nor externally.
+
+#### Upgrade a multi-node cluster
+
+As a technician, I want to be able to upgrade a multi-node cluster using the upgrade package and
+without requiring/deploying an image registry server neither on the cluster itself nor externally.
+
+### Goals
+
+Provide a documented and supported mechanism that technicians can use to upgrade a cluster without
+requiring a registry server.
+
+### Non-Goals
+
+It is not a goal to not require a registry server for other operations. For example, installing a
+new workload will still require a registry server.
+
+It is not a goal to provide a tool to automate the procedures to prepare the upgrade package or to
+perform the upgrade. Such an automation will be part of a future enhancement.
+
+### Workflow Description
+
+1. An engineer working in a partner's factory is asked to prepare a package to upgrade a cluster to
+a specific OpenShift architecture and version.
+
+1. The engineer uses the documentation and tools to prepare the upgrade package containing all the
+artifacts (mostly release images) that are required to perform the upgrade and writes it to some
+kind of media (USB stick, for example) that will be delivered to the technicians.
+
+1. The technicians receive copies of the upgrade package.
+
+1. The technician goes to the cluster location and uses the upgrade package, documentation and tools
+to perform the upgrade. This step is potentially repeated multiple times for multiple clusters.
+
+Note that the upgrade package should not be specific for a particular cluster, only for the
+OpenShift architecture and version. Technicians should be able to use that package for any cluster
+with that architecture and version.
+
+### API Extensions
+
+None.
+
+### Implementation Details/Notes/Constraints
+
+The proposed solution is based on copying all the required images to all the nodes of the cluster
+before starting the upgrade, and ensuring that no component requires access to a registry server
+during the upgrade. For this to work the following changes are required:
+
+1. No OpenShift component used during the upgrade should use the `Always` pull policy, as that
+forces the kubelet and CRI-O to try to contact the registry server even if the image is already
+available.
+
+1. No OpenShift component should garbage collect the upgrade images before or during the upgrade.
```

Comment:
This is  typically started by the kubelet instructing CRI-O to remove images.

Reply:
Done.